### DISCUSSION TITLE
Flow speed and RBC generation

### DISCUSSION THREAD

**Discussion by jounihuo** [UvaCsl/HemoCell#66] [General discussion]
I have modified the pipeflow example to have a simple time dependent flow speed amplitude.

` setExternalVector(*hemocell.lattice, hemocell.lattice->getBoundingBox(),
                DESCRIPTOR<T>::ExternalField::forceBeginsAt,
                plb::Array<T, DESCRIPTOR<T>::d>(poiseuilleForce*amplitude, 0.0, 0.0));`

But do I need to take steps to ensure RBC generation has similar amplitude (if yes then how?) or is it automatic?

**Comment by gzavo** [comment]
Hi @jounihuo,

If you have a periodic simulation (as in the `pipeflow` case), the number of cells are constant during the run, so you only need to reset the driving force in every iteration, the cell generation happens using the `cellPack` tool before running the simulation.
If you have a more complex domain, that is not periodic (e.g., a capillary network), then you can use the periodic pre-inlet to drive the domain. See the documentation [here](https://www.hemocell.eu/user_guide/cases/pipeflow_with_preinlet.html), and related publications describing it [here](https://www.sciencedirect.com/science/article/pii/S0045793018302214) and applying it [here](https://www.tandfonline.com/doi/pdf/10.1080/10255842.2022.2034794).
